### ISSUE TITLE
How do I transer save files between versions? 

### ISSUE THREAD

**Issue by kurayami6** [stenzek/duckstation#327]
Forgive me for being dumb but can anyone explain it, please?

**Comment by stenzek** [comment]
I'm not sure what you're asking.

Save states are not compatible when the structure changes. This isn't every version, but does happen periodically.

When you save your game using the ingame save functionality, it writes it to the memory card image in the memcards directory. this is a raw image of a psx memory card, so it will always work no matter what version you use.

When you update, extract the archive to the same directory you used before. This will preserve your memory cards and settings.

Tldr; use memory cards. Save states change between versions in most emulators so you can't always rely on them and should save to memory card where possible.

**Comment by kurayami6** [comment]
Thanks for the explanation. My problem was I was downloading a new version and copying the memory card from the previous version to the new one and it wasn't working. I know I've already asked for it but an auto updater would make things a lot simpler.

**Comment by stenzek** [comment]
Hm, not really sure why copying the card over wouldn't work. Unless it wasn't in the same directory.

But really, just extract the new build over the top of the old one and overwrite it. It's not like the core exes/dlls are changing, so you won't have stray files which aren't necessary, and you preserve your settings this way. The only reason to keep multiple builds would be if you were regression testing.

I might consider storing the data in `My Documents` instead though. That would solve this issue entirely as it wouldn't be dependent on where you ran it from (while keeping the portable option of course).

**Comment by kurayami6** [comment]
That's all I was confused about. I would copy the memory card from the old version to the new one and the game would say the memory card was empty. I assumed I was doing something wrong.

**Comment by stenzek** [comment]
Still strange. It should use the existing file if it's there.

Alternatively, I guess you could store your memory cards elsewhere and change the path in settings to point to that instead.

**Comment by kurayami6** [comment]
I just did what you said, extracted the new version in the same directory so it overwrote the old version and it's still saying the card is empty.

![Capture](https://user-images.githubusercontent.com/31968468/80122330-bdaaca80-85d0-11ea-81a7-aebaf4a4d081.PNG)

**Comment by stenzek** [comment]
Potentially silly question, but did you save the game to the memory card first?

**Comment by kurayami6** [comment]
Yes of course. I don't like using save states because it feels like cheating.

**Comment by stenzek** [comment]
I'm really not sure what's going on then. I just tried using a different build in the same folder and it loaded the memory card from the first one fine, as well as copying the `mcd` file to a different folder and running it out of that.

Make sure you're writing for the memcard to flush before powering off the system - this is when the "memory card xyz saved" message is displayed, usually a few seconds after the game writes to the card.

Also check the console/log to make sure there's no messages about memory cards being formatted, or failing to load. Because if it isn't loading, that would explain what's going on.
